src/BatteryStateOfChargeService.h

```diff
@@ -42,8 +42,16 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     void addData(const BatteryData& batteryData);
 
 private:
-    // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
 
-    // Recommend adding necessary variables here
+    QTime previousTime_;
+    QTime return_time;
+
+    double avgCurrent_;
+    double previousCurrent_;
+
+    double deltaTime_;
+
+    double totalAH_;
+    double deltaAH_;
```

Comment:
This variable can just be a local variable in `addData`, doesn't need to be a private member. In general, you want to give a variable as small a scope as possible.